include/camp/resource.hpp

```diff
@@ -94,29 +94,74 @@ namespace resources
       void wait_for(Event *e) { m_value->wait_for(e); }
       void wait() { m_value->wait(); }
 
+      /*
+       * \brief Compares two Resources to see if they are equal. Two Resources are equal if they are
+       * on the same platform and they describe the same stream (i.e. CUDA, HIP) or queue (i.e. SYCL).
+       * (Note: This operator overload is on the generic Resource. Using the ContextModel's operator
+       * overload, we can call the specific resource's (i.e. T) operator== to compare streams or
+       * queues.)
+       *
+       * \return True or false depending on comparison with m_value if they are on the same platform.
+       */
+      bool operator==(Resource const& r) const
+      {
+        if(get_platform() == r.get_platform()) {
+          return (m_value->compare(r));
+        }
+        return false;
+      }
+      /*
+       * \brief Compares two Resources to see if they are NOT equal. Two Resources are not equal
+       * if they are on separate platforms. Also, even if they are on the same platform, if they
+       * describe different streams (i.e. CUDA, HIP) or different queues (i.e. SYCL), then they 
+       * are not equal.
+       *
+       * \return Negation of == operator. 
+       */
+      bool operator!=(Resource const& r) const
+      {
+        return !(*this == r);
+      }
+
+      template <typename T>
```

Comment:
You'll have to disable this method when T is Resource to avoid conflicting with the other `operator==`.
This also only allows you to compare with a concrete resource on the right hand side of an ==, to allow comparison with a concrete resource on the left hand side you'll have to add a namespace scope operator== between T and Resource.
```suggestion
      template <typename T, std::enable_if_t<std::is_same<T, Resource>::value>* = nullptr>
```

Reply:
Wait, that looks backwards @MrBurmark, it needs to be enabled when T is _not_ same right?

Reply:
thank you both!